### PR TITLE
Formatting internal documentation and adding table of contents

### PR DESCRIPTION
**Pull request by sanketsudake**
Changes format internal READMEs for better readability and added a table of contents.
Using one sentence per one line improves readability and makes editing easier, also helps in reducing no. of complex statements.

### CODE REVIEW

File: Documentation/Architecture.md [hunk 1]
```diff
@@ -1,148 +1,140 @@
...
-the HPA scales the replicas of the deployment based on the conditions set by the user.
- 
-This approach though increases the cold time of a function, but also makes NewDeploy 
-suitable for functions designed to serve massive traffic.
+PoolManager watches the environment CRD changes and eagerly creates generic pools for environments.
+It uses Kubernetes deployments to do that.
+The environment container runs in a pod with the 'fetcher' container.
+Fetcher is a straightforward utility that downloads a URL sent to it and saves it at a configured location (shared volume).
+
+The implementation chooses a generic pod from the pool, relabels it to "orphan".
```
Comment (vishal-biyani) [inline Documentation/Architecture.md:87]:
```suggestion
The implementation chooses a generic pod from the pool, relabels it by adding `functionName` label and `managed=false` label.
```

File: Documentation/Architecture.md [hunk 2]
```diff
@@ -1,148 +1,140 @@
...
+The pod from the deployment invokes fetcher to copy the function into the pod and hits the specialize endpoint on the environment container.
+This causes the function to be loaded.
+The pod is now specific to that function.
+This function pod is cached; it's cleaned up if it's unused for a few minutes.
+
+PoolManager selects a generic pod from the warm pool, specializes it, and recycles the pod if there are no further requests to the function after a few minutes.
+It makes PoolManager suitable for functions that are short-living and requires a short cold start time [1].
+
+However, PoolManager only selects one pod per function, which is not suitable for serving massive traffic.
```
Comment (vishal-biyani) [inline Documentation/Architecture.md:96]: This is not 100% true with RPP. Also Concurrency feature adds some more scenarios, but may be for some other PR

**Review by vishal-biyani — CHANGES_REQUESTED** [review]
Minor comments

File: Documentation/Architecture.md [hunk 3]
```diff
@@ -154,47 +146,37 @@ The fetcher inside the function pod also pulls the deploy archive for function s
...
-via the router.
+The controller keeps track of the user's requested watches and associated functions.
+Kubewatcher watches the API based on these requests; when a watch event occurs, it serializes the object and calls the function via the router.
 
-While a few simple retries are done, there isn't yet a reliable
-message bus between Kubewatcher and the function. Work for this is
-tracked in issue #64.
+While a few simple retries are done, there isn't yet a reliable message bus between Kubewatcher and the function. Work for this is tracked in issue #64.
 
 ### Message Queue Trigger
```
Comment (RealHarshThakur) [inline Documentation/Architecture.md:167]: Can we also add about KEDA in this section?

**Pull request closed without merging**